ddclient.in

```diff
@@ -2155,6 +2168,171 @@ sub geturl {
     return $reply;
 }
 
+######################################################################
+## curl_cmd() function to execute system curl command
+######################################################################
+sub curl_cmd {
+    my @params = @_;
+    my $tmpfile;
+    my $tfh;
+    my $system_curl = subst_var('@syscurlcmd@', '/usr/bin/curl');
```

Comment:
No, there should be a default just like @sysconfdir@ and @localstatedir@ so that it does not error out.

Reply:
Sorry, I wasn't clear. What I mean is this:

```perl
    my $system_curl = subst_var('@CURL@', '/usr/bin/curl');
    fatal("curl not found") if ($system_curl eq '');
```

This way, if `@CURL@` is replaced with the empty string it won't try to execute the empty string.

And now that I think about it more, perhaps we should default to a non-absolute `curl` pathname in `configure.ac` so that it can be found via a `PATH` search:

```autoconf
AC_PATH_PROG([CURL], [curl], [curl])
```